README.md

```diff
@@ -8,22 +11,84 @@ If you want a module that specifically aims to be a complete and transparent int
 
 Initially this module was named Net::LDNS.
 
-IDN
-===
-If GNU libidn is installed when this module is compiled, it will be used to add a simple function that converts strings from Perl's internal encoding to IDNA domain name format. In order to convert strings from whatever encoding you have to Perl's internal format, use L<Encode>. If you need any kind of control or options, use L<Net::LibIDN>. The included function here is only meant to assist in the most basic case, although that should cover a lot of real-world use cases.
 
-Installation
-============
+## Features
+
+When installing from source, you can choose to enable or disable a number
+of optional features using command line options to the `perl Makefile.PL`
+commands.
+
+### IDN
+
+This feature is enabled by default.
+Disable with `--no-idn`.
+
+If the IDN feature is enabled, the GNU `libidn` library will be used to
+add a simple function that converts strings from Perl's internal encoding
+to IDNA domain name format.
+In order to convert strings from whatever encoding you have to Perl's
+internal format, use L<Encode>.
+If you need any kind of control or options, use L<Net::LibIDN>.
+The included function here is only meant to assist in the most basic case,
+although that should cover a lot of real-world use cases.
+
+> **Note:** The Zonemaster Engine test suite assumes this feature
+> is enabled.
+
+### Randomized capitalization
+
+This feature is disabled by default.
+Enable with `--randomize`.
+
+Randomizes the capitalization of returned domain names.
+
+
+## Dependencies
+
+This module depends on the `openssl` library and has an optional
+dependency on the `libidn` (with the [IDN](#idn) feature, enabled by
+default) library.
+
```

Comment:
Until we understand what this feature does and if it can create any harm if enabled, add:

> **Note:** The Zonemaster Engine test suite assumes this feature is disabled.